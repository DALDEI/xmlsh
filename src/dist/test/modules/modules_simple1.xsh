# Test simple modules
. ../common

# Simple script module
XMODPATH=(. ../../samples/scripts $XMODPATH)
import s=simple_module
# test A Test
echo  -n loc()
s:run A Test
echo  -n loc()
s:run2
echo  -n loc()
: s:run2()

A=parent
echo  -n loc()
s:seta AText

echo loc() A is $A
echo  -n loc()
s:setc CText
echo loc() C is $C
unset A
s:seta ATextMore
echo $A
unset C
s:setc CTextMore
echo loc() parent C is $C
echo loc() internal C is s:getc() 
s:run2
echo  -n loc()
: s:run_child( Child text )
C=ParentC
echo  -n loc()
: s:ev( 'echo $C' )
echo loc() child local: s:getlocal()
: s:setlocal( parent local) 
echo loc() child local: s:getlocal()
x=s
echo loc() $x:varf( f1 f1 args )
y=varf
echo loc() $x:$y( f2 f2 args )

