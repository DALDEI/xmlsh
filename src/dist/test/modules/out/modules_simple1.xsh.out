initializing modules_simple1.xsh
[8]test A Test
[10]A is
B is
C is bar
[12]A is
B is
C is bar
[16]A is parent
A is AText
[19] A is AText
[20]C is bar
C is CText
[22] C is
A is
A is ATextMore
ATextMore
C is bar
C is CTextMore
[28] parent C is
[29] internal C is bar
A is ATextMore
B is
C is bar
[31]child Child text
[34]bar
[36] child local: 123
[38] child local: 123
[40] f1 args
[42] f2 args
